pkg/providers/aws/provider_redis.go

```diff
@@ -255,26 +255,30 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 		}
 		return nil, "started elasticache provision", nil
 	}
-
-	// check elasticache phase
-	if *foundCache.Status != "available" {
-		logger.Infof("found instance %s current status %s", *foundCache.ReplicationGroupId, *foundCache.Status)
-		return nil, croType.StatusMessage(fmt.Sprintf("createReplicationGroup() in progress, current aws elasticache status is %s", *foundCache.Status)), nil
-	}
-	logger.Infof("found existing elasticache cluster %s", *foundCache.ReplicationGroupId)
-
 	cacheClustersOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{})
 	if err != nil {
 		errMsg := "failed to describe clusters"
 		return nil, croType.StatusMessage(fmt.Sprintf(errMsg)), errorUtil.Wrapf(err, errMsg)
 	}
 	var replicationGroupClusters []elasticache.CacheCluster
 	for _, checkedCluster := range cacheClustersOutput.CacheClusters {
-		if *checkedCluster.ReplicationGroupId == *foundCache.ReplicationGroupId {
-			replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
+		if *checkedCluster.ReplicationGroupId != "" && *foundCache.ReplicationGroupId != "" {
```

Comment:
```suggestion
		if checkedCluster.ReplicationGroupId != nil && foundCache.ReplicationGroupId != nil {
```

Reply:
removing this nil check as it conflicts with changes @laurafitzgerald Laura has done on master

Reply:
looks like the unit tests are failing without this  
![image](https://user-images.githubusercontent.com/16667688/146008848-3e6f7ae8-d08f-4b86-9ccf-07df8e8ae23b.png)